File: 35-search-insert-position/35-search-insert-position.c
int searchInsert(int* nums, int numsSize, int target){
    int l,r,mid;
    l = 0;
    r = numsSize-1;
    
    while(l <= r)
    {
        mid = (l+r)/2;
        if(target > nums[numsSize-1])
            return numsSize;
        else if(nums[mid] == target)
            return mid;
        else if(target > nums[mid] && target < nums[mid+1])
            return mid+1;
        else if(target < nums[mid])
            r = mid-1;
        else if(target > nums[mid])
            l = mid+1;
    }
    return 0;
}

File: 35-search-insert-position/35-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int l = 0;
        int r = nums.length-1;
        
        while(l <= r)
        {
            int mid = (l + r) / 2;
            if(nums[mid] == target)
                return mid;
            else if(mid+1 < nums.length && target > nums[mid] && nums[mid+1] > target)
                return mid+1;
            else if(target > nums[mid] && mid+1 == nums.length)
                return mid+1;
            else if(nums[mid] > target)
                r = mid-1;
            else
                l = mid+1;
        }
        return 0;
    }
}